datavault/backend.py

```diff
@@ -216,7 +439,378 @@ def hasMore(self, pos):
+                dtype.append((varname, shapestr + 'c16'))
+            else:
+                raise RuntimeError("Invalid type tag {}".format(ttag))
+
+        self.file.create_dataset('DataVault', (0,), dtype=dtype, maxshape=(None,))
+        HDF5MetaData.initialize_info(self, title, indep, dep)
+
+    @property
+    def file(self):
+        return self._file()
+    @property
+    def dataset(self):
+        return self.file["DataVault"]
+
+    def addData(self, data):
+        """Adds one or more rows or data from a numpy struct array."""
+        new_rows = len(data)
+        old_rows = self.dataset.shape[0]
+        self.dataset.resize((old_rows + new_rows,))
+        self.dataset[old_rows:(old_rows + new_rows)] = data
+
+    def getData(self, limit, start, transpose, simpleOnly):
+        """Get up to limit rows from a dataset."""
+        if simpleOnly:
+            datatype = self.dataset.dtype
+            for idx in range(len(datatype)):
+                if datatype[idx] != np.float64:
+                    raise errors.DataVersionMismatchError()
+        if transpose:
+            return self.getDataTranspose(limit, start)
+
+        data, new_pos = self._getData(limit, start)
+        row_data = [tuple(row) for row in data]
+        return row_data, new_pos
+
+    def getDataTranspose(self, limit, start):
+        struct_data, new_pos = self._getData(limit, start)
+        columns = []
+        for idx in range(len(struct_data.dtype)):
+            col = struct_data['f{}'.format(idx)]
+            # Strings are stored as hdf5 vlen objects.  Numpy can't do
+            # variable length strings, so they get encoded as object
+            # arrays by hdf5.  we don't know how to flatten object
+            # arrays so we special case vlen types here and convert
+            # them to lists.  Also, h5py has a bug where when you
+            # index a dataset with a compound type, it loses the
+            # special dtype information, so we pull it directly from
+            # self.dataset.dtype rather than the data returned by
+            # _getData
+            if self.dataset.dtype[idx] == np.object:
+                base_type = h5py.check_dtype(vlen=self.dataset.dtype[idx])
+                if not base_type or not issubclass(base_type, str):
+                    raise RuntimeError("Found object type array, but not vlen str.  Not supported.  This shouldn't happen")
+                col = [base_type(x) for x in col]
+            columns.append(col)
+        columns = tuple(columns)
+        return columns, new_pos
+
+    def _getData(self, limit, start):
+        if limit is None:
+            struct_data = self.dataset[start:]
+        else:
+            struct_data = self.dataset[start:start+limit]
+        return struct_data, start + struct_data.shape[0]
+
+    def __len__(self):
+        return self.dataset.shape[0]
+
+    def hasMore(self, pos):
+        return pos < len(self)
+
+class SimpleHDF5Data(HDF5MetaData):
```

Comment:
Is this not just a special case of HDF5MetaData? Why do we need a whole class? Is it just for backwards compatibility?